pkg/vm/createEvm.go

```diff
@@ -73,54 +54,48 @@ func CreateEvmSubnetConfig(app *application.Avalanche, subnetName string, genesi
 	return genesisBytes, sc, nil
 }
 
-func nextStage(currentState wizardState, direction stateDirection) wizardState {
-	switch direction {
-	case forward:
-		currentState++
-	case backward:
-		currentState--
-	default:
-		return errored
-	}
-	return currentState
-}
-
 func createEvmGenesis(app *application.Avalanche, subnetName string, subnetEVMVersion string) ([]byte, *models.Sidecar, error) {
 	ux.Logger.PrintToUser("creating subnet %s", subnetName)
 
 	genesis := core.Genesis{}
 	conf := params.SubnetEVMDefaultChainConfig
 
-	stage := startStage
+	const (
+		descriptorsState = "descriptors"
+		feeState         = "fee"
+		airdropState     = "airdrop"
+		precompilesState = "precompiles"
+	)
 
 	var (
 		chainID    *big.Int
 		tokenName  string
 		vmVersion  string
 		allocation core.GenesisAlloc
-		direction  stateDirection
+		direction  statemachine.StateDirection
 		err        error
 	)
 
-	for stage != doneStage {
-		switch stage {
-		case startStage:
-			direction = forward
-		case descriptorStage:
+	subnetEvmState := statemachine.NewStateMachine(
+		[]string{descriptorsState, feeState, airdropState, precompilesState},
+	)
+	for subnetEvmState.Running() {
```

Comment:
It looks like there is another invariant, which is that the state direction always gets automatically initialized to `Forward` because it takes the null value if the var is not initialized.

This is clear looking at the code, but for a user of the `StateMachine` it might be also worth documenting somewhere.

Reply:
state direction is controlled by the user, and passed using NextState(). is initialized to whatever the module user wants.

Reply:
Yes and in this case the user did not initialized by the user (at least I did not see it).

Which means there is an implicit assumption that it starts with `Forward` if nothing has been initialized?